packages/ra-core/src/util/useDebouncedEvent.ts

```diff
@@ -0,0 +1,19 @@
+import { useCallback, useRef, useEffect } from 'react';
+import debounce from 'lodash/debounce';
+
+// Hook somewhat equivalent to useEvent, but with a debounce
+// Returns a debounced callback which will not change across re-renders unless the
+// callback or delay changes
+export const useDebouncedEvent = (callback, delay: number) => {
+    // Create a ref that stores the debounced callback
+    const debouncedCallbackRef = useRef(debounce(callback, delay));
+
+    // Whenever callback or delay changes, we need to update the debounced callback
+    useEffect(() => {
+        debouncedCallbackRef.current = debounce(callback, delay);
+    }, [callback, delay]);
+
+    // The function returned by useCallback will invoke the debounced callback
+    // Its dependencies array is empty, so it never changes across re-renders
+    return useCallback((...args) => debouncedCallbackRef.current(...args), []);
+};
```

Comment:
This doesn't have the checks we do in `useEvent` to prevent calling the callback inside a render. I'm not really sure of the consequences but the react team has made it a strong limitation of `useEvent` (see its code in our repo)